src/prog_models/sim_result.py

```diff
@@ -32,7 +33,25 @@ def __init__(self, times : list = None, data : list = None, _copy = True):
             else:
                 self.data = data
 
-    def __eq__(self, other : "SimResult") -> bool:
+    @property
+    def frame(self) -> pd.DataFrame:
+        """
+        Getter -- creating multi row pd.DataFrame from data list of dict
+
+            Return: pd.DataFrame, of all the data and times
+        """
+        if len(self.data) > 0:  #
+            self._frame = pd.concat([
```

Comment:
What is the purpose of _frame as a property instead of a local variable? _frame is recalculated every time you access frame. Are you planning on using caching in a future version.

Reply:
No cach in the future, the reason was so that people had the option for the view if they wanted. Since at this stage DF aren't being used in any of the object functions, creating a pandas object every time will just slow down the use of sim_result when it's being forced to convert from data being run without any pandas support. When progpy runs on pandas then it would make sense for sim_result to be fully in pandas as well and no need for the property